activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -174,6 +189,21 @@ def setup_transactional_fixtures
 
         # Begin transactions for connections already established
         @fixture_connection_pools = ActiveRecord::Base.connection_handler.connection_pool_list(:writing)
+
+        transaction_disabled_dbnames = database_transactions_config.select { |_, enabled| !enabled }.keys
+        transaction_enabled_dbnames = database_transactions_config.select { |_, enabled| enabled }.keys
+
+        # Remove any pools that we don't want to use with transactions
+        @fixture_connection_pools.reject! { |pool|
+          if use_transactional_tests
+            # If the default is to use transactions, remove pools that are explicitly DISabled
+            transaction_disabled_dbnames.include?(pool.db_config.name.to_sym)
+          else
+            # Otherwise the default is NOT to use transactions, so remove any pools that are NOT explicitly ENabled
+            !transaction_enabled_dbnames.include?(pool.db_config.name.to_sym)
+          end
+        }
```

Comment:
```suggestion
        # Filter to pools that want to use transactions
        @fixture_connection_pools.select! { |pool|
          database_transactions_config.fetch(pool.db_config.name.to_sym, use_transactional_tests)
        }
```

(Is that the right/best way to get the name symbol? Feels clunky.)